atsign-foundation/at_client_sdk
# fix: Skip cached public keys and cached shared keys from enrollment a…

…uthorization check



**- What I did**
 - When authenticated with APKAM, operations performed in a local secondary database exclude only reserved keys from validation. However, there is an issue with validating the cached `sharedkey`and `encryption public key`because these keys lack a namespace for validation.
 - Along with the reserved keys, exclude the cached shared_key and cached public key from the enrollment authorization check.

**- How I did it**
In local_secondary.dart, modify the existing method which excludes the local keys from enrollment authorization check to also exclude the cached `sharedkey` and `encryption public key`.

**- How to verify it**
- All the existing unit and functional test should pass.
- Tested the no-ports E2E tests locally with APKAM authentication.

Attaching the test logs
[srvd.log](https://github.com/user-attachments/files/15787816/srvd.log)
[sshnp.log](https://github.com/user-attachments/files/15787817/sshnp.log)
[sshnpd.log](https://github.com/user-attachments/files/15787818/sshnpd.log)

Pending work:
- Add unit tests to ensure the cached shared key, cached public key and reserved keys are excluded from enrollment authorization check. Created a git issue and working on it : https://github.com/atsign-foundation/at_client_sdk/issues/1335
